vlsi/Makefile

```diff
@@ -15,108 +15,91 @@ sim_dir=$(abspath .)
+# This should be built alongside $(SMEMS_FILE)
+$(SMEMS_HAMMER): $(SMEMS_FILE)
 
+$(SRAM_GENERATOR_CONF): $(SMEMS_HAMMER)
+	mkdir -p $(dir $@)
+	echo "vlsi.inputs.sram_parameters: '$(SMEMS_HAMMER)'" >> $@
+	echo "vlsi.inputs.sram_parameters_meta: [\"transclude\", \"json2list\"]">> $@
 
-###################################################### SYN ############################################################
+$(SRAM_CONF): $(SRAM_GENERATOR_CONF)
+	cd $(vlsi_dir) && $(HAMMER_EXEC) -e $(ENV_YML) $(foreach x,$(INPUT_CONFS) $(SRAM_GENERATOR_CONF), -p $(x)) --obj_dir $(build_dir) sram_generator
+	cd $(vlsi_dir) && cp output.json $@
 
-SYNTH_CONF = $(OBJ_DIR)/inputs.yml
+#########################################################################################
+# synthesis input configuration
+#########################################################################################
+SYN_CONF = $(OBJ_DIR)/inputs.yml
+GENERATED_CONFS = $(SYN_CONF) $(SRAM_CONF)
 
-$(SYNTH_CONF):
+$(SYN_CONF): $(ALL_RTL) $(extra_v_includes) $(sim_top_blackboxes)
 	mkdir -p $(dir $@)
 	echo "synthesis.inputs:" > $@
 	echo "  top_module: $(TOP)" >> $@
 	echo "  input_files:" >> $@
-	for x in $(ALL_RTL); do \
+	for x in $(ALL_RTL) $(extra_v_includes) `cat $(sim_top_blackboxes)`; do \
 		echo '    - "'$$x'"' >> $@; \
 	done
 
-GENERATED_CONFS=$(SYNTH_CONF) $(SRAM_CONF)
-
-.PHONY: syn synthesis
-syn: $(OBJ_DIR)/syn-rundir/$(TOP).mapped.v
-synthesis: syn
-
-$(OBJ_DIR)/syn-rundir/$(TOP).mapped.v $(OBJ_DIR)/syn-rundir/syn-output.json: $(ENV_YML) $(INPUT_CONFS) $(GENERATED_CONFS) $(ALL_RTL)
-	mkdir -p $(dir $@)
-	$(HAMMER_EXEC) -e $(ENV_YML) $(foreach x,$(INPUT_CONFS) $(GENERATED_CONFS), -p $(x)) --obj_dir $(OBJ_DIR) syn
-
-$(OBJ_DIR)/par-input.json: $(OBJ_DIR)/syn-rundir/syn-output.json
-	mkdir -p $(dir $@)
-	$(HAMMER_EXEC) -e $(ENV_YML) $(foreach x,$(INPUT_CONFS) $<, -p $(x)) -o $@ syn_to_par
-
-###################################################### PAR ############################################################
-
-.PHONY: par place-and-route
-par: $(OBJ_DIR)/par-rundir/$(TOP).gds
-place-and-route: par
-
-$(OBJ_DIR)/par-rundir/$(TOP).gds $(OBJ_DIR)/par-rundir/par-output.json: $(OBJ_DIR)/par-input.json $(OBJ_DIR)/syn-rundir/$(TOP).mapped.v
-	mkdir -p $(dir $@)
-	$(HAMMER_EXEC) -e $(ENV_YML) -p $< --obj_dir $(OBJ_DIR) par
-
-$(OBJ_DIR)/drc-input.json: $(OBJ_DIR)/par-input.json $(OBJ_DIR)/par-rundir/par-output.json
-	mkdir -p $(dir $@)
-	$(HAMMER_EXEC) -e $(ENV_YML) $(foreach x,$^, -p $(x)) -o $@ --obj_dir $(OBJ_DIR) par_to_drc
+#########################################################################################
+# AUTO BUILD FLOW
+#########################################################################################
 
-###################################################### DRC ############################################################
-# TODO unimplemented
-.PHONY: drc
-drc: $(OBJ_DIR)/drc-rundir/drc_results.db
+.PHONY: buildfile
+buildfile: $(OBJ_DIR)/hammer.d
 
```

Comment:
```suggestion
# Tip: Set HAMMER_D_DEPS to an empty string to avoid unnecessary RTL rebuilds
# TODO: make this dependency smarter so that we don't need this at all
HAMMER_D_DEPS ?= $(GENERATED_CONFS)
```